### ISSUE TITLE
Add Megatron support for the EleutherAI Evaluation Harness

### ISSUE THREAD

**Issue by ibeltagy** [bigscience-workshop/Megatron-DeepSpeed#137]
Add the ability to run the EleutherAI Evaluation Harness on Megatron checkpoints. Right now we are relying on converting Megatron checkpoints to Hugginface checkpoints which is an error-prone process. We also have to use Megatron anyways to run the 200B model. 

Implementation details: 
You will use this HF gpt2 model implementation [here](https://github.com/EleutherAI/lm-evaluation-harness/blob/master/lm_eval/models/gpt2.py) as your reference. Here are more details:
- Edit `_init_, create_from_arg_string` to load the Megatron checkpoints
- Edit the `_model_call` function to call the Megatron model and read logits back
- The functions `loglikelihood, loglikelihoods, _loglikelihood_tokens` might (or might not) require a little tweaking
- Leave the function `greedy_until` unimplemented (raise an exception), we don't need it for now. 
- Check [this test](https://github.com/bigscience-workshop/Megatron-DeepSpeed/blob/43881ff5506a0e3bd3c4937d375030addeea140a/tests/test_conversion.py#L123-L159) that shows how to load and call a Megatron checkpoint.
- [Here's](https://huggingface.co/bigscience/gpt2-350m-en/tree/megatron-deepspeed) one Megatron checkpoint that you can work with.
- A relatively close implementation is already in the GPT-NeoX repo [here](https://github.com/EleutherAI/gpt-neox/blob/main/eval_tasks/adaptor.py) and it might be helpful to check as well.

**Comment by StellaAthena** [comment]
@ibeltagy Ooops, looks like a misclicked. Thanks
